components/registry-facade/cmd/run.go

```diff
@@ -134,3 +138,75 @@ func authorizerFromDockerConfig(cfg *configfile.ConfigFile) docker.Authorizer {
 		return
 	}))
 }
+
+// watchConfig watches the configuration file and if changed reloads the static layer
+func watchConfig(fn string, reg *registry.Registry) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.WithError(err).Error("cannot watch config - config hot reloading will not work")
+		return
+	}
+	defer watcher.Close()
+
+	err = watcher.Add(fn)
+	if err != nil {
+		log.WithError(err).Error("cannot watch config - config hot reloading will not work")
+		return
+	}
+
+	defer log.Error("config watching has ended")
+
+	reloadConfig := func() error {
+		cfg, err := getConfig(fn)
+		if err != nil {
+			return err
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		return reg.UpdateStaticLayer(ctx, cfg.Registry.StaticLayer)
+	}
+
+	for {
+		select {
+		case event, ok := <-watcher.Events:
+			if !ok {
+				return
+			}
+
+			log.WithField("event", event).Debug("config event")
+			var reload bool
+
+			// k8s configmaps uses symlinks, we need this workaround.
+			// original configmap file is removed
+			if event.Op == fsnotify.Remove {
+				// remove the watcher since the file is removed
+				watcher.Remove(event.Name)
+				// add a new watcher pointing to the new symlink/file
+				watcher.Add(fn)
+
+				reload = true
+			}
+
+			if event.Op&fsnotify.Write == fsnotify.Write {
+				reload = true
+			}
+
+			if reload {
+				err = reloadConfig()
+				if err == nil {
+					log.Warn("configuration was updated - reloaded static layer config")
```

Comment:
I'd prefer info over warning as reloading the configuration is intended behaviour.